```
checking for promoted mode_t type... libvirt-glib.c:16:20: fatal error: Python.h: No such file or directory
 #include <Python.h>
                    ^
compilation terminated.
```

--- a/pkgs/development/libraries/libvirt-glib/default.nix
+++ b/pkgs/development/libraries/libvirt-glib/default.nix
@@ -1,35 +1,20 @@
 { stdenv, fetchurl, pkgconfig, libvirt, glib, libxml2, intltool, libtool, yajl
 , nettle, libgcrypt, python, pygobject, gobjectIntrospection, libcap_ng, numactl
+, xen
 }:
 
 stdenv.mkDerivation rec {
-  name = "libvirt-glib-0.2.0";
+  name = "libvirt-glib-0.2.3";
 
   src = fetchurl {
     url = "http://libvirt.org/sources/glib/${name}.tar.gz";
-    sha256 = "02saqkk4wzsimsan7s9yc5bx05xn7j00hnxhq4sczkgr4krf1drh";
+    sha256 = "1pahj8qa7k2307sd57rwqwq1hijya02v0sxk91hl3cw48niimcf3";
   };
 
   buildInputs = [
     pkgconfig libvirt glib libxml2 intltool libtool yajl nettle libgcrypt
-    python pygobject gobjectIntrospection libcap_ng numactl
+    python pygobject gobjectIntrospection libcap_ng numactl xen
   ];
-
-  # Compiler flag -fstack-protector-all fixes this build error:
-  #
-  #   ./.libs/libvirt-glib-1.0.so: undefined reference to `__stack_chk_guard'
-  #
-  # And the extra include path fixes this build error:
-  #
-  #   In file included from ../libvirt-gobject/libvirt-gobject-domain-device.h:30:0,
-  #                    from /tmp/nix-build-libvirt-glib-0.1.7.drv-2/libvirt-glib-0.1.7/libvirt-gobject/libvirt-gobject.h:33,
-  #                    from <stdin>:4:
-  #   ../libvirt-gobject/libvirt-gobject-domain.h:33:29: fatal error: libvirt/libvirt.h: No such file or directory
-  #   compilation terminated.
-  #   make[3]: *** [LibvirtGObject-1.0.gir] Error 1
-  preConfigure = ''
-    export NIX_CFLAGS_COMPILE="-fstack-protector-all -I${libvirt}/include"
-  '';
 
   meta = with stdenv.lib; {
     description = "Library for working with virtual machines";

--- a/pkgs/development/libraries/libvirt/default.nix
+++ b/pkgs/development/libraries/libvirt/default.nix
@@ -2,7 +2,7 @@
 , iproute, iptables, readline, lvm2, utillinux, udev, libpciaccess, gettext
 , libtasn1, ebtables, libgcrypt, yajl, makeWrapper, pmutils, libcap_ng
 , dnsmasq, libnl, libpcap, libxslt, xhtml1, numad, numactl, perlPackages
-, curl, libiconv, gmp
+, curl, libiconv, gmp, xen
 }:
 
 stdenv.mkDerivation rec {
@@ -19,7 +19,8 @@ stdenv.mkDerivation rec {
     gettext libtasn1 libgcrypt yajl makeWrapper
     libxslt xhtml1 perlPackages.XMLXPath curl libpcap
   ] ++ stdenv.lib.optionals stdenv.isLinux [
-    libpciaccess devicemapper lvm2 utillinux udev libcap_ng libnl numad numactl
+    libpciaccess devicemapper lvm2 utillinux udev libcap_ng
+    libnl numad numactl xen
   ] ++ stdenv.lib.optionals stdenv.isDarwin [
      libiconv gmp
   ];

--- a/pkgs/development/libraries/libvirt/default.nix
+++ b/pkgs/development/libraries/libvirt/default.nix
@@ -17,7 +17,7 @@ stdenv.mkDerivation rec {
   buildInputs = [
     pkgconfig libxml2 gnutls perl python readline
     gettext libtasn1 libgcrypt yajl makeWrapper
-    libxslt xhtml1 perlPackages.XMLXPath curl
+    libxslt xhtml1 perlPackages.XMLXPath curl libpcap
   ] ++ stdenv.lib.optionals stdenv.isLinux [
     libpciaccess devicemapper lvm2 utillinux udev libcap_ng libnl numad numactl
   ] ++ stdenv.lib.optionals stdenv.isDarwin [

